### PR TITLE
Add Get and Post for ProgressReport

### PR DESCRIPTION
**Pull request by vegarsti**
This PR adds Get and Post methods for the progress report API (https://github.com/duneanalytics/duneapi/pull/559).

We use the endpoints in #17 (and they work!).

Towards CHAIN-1588.

### CODE REVIEW

**Comment by vegarsti** [comment]
* **#16** <a href="https://app.graphite.dev/github/pr/duneanalytics/node-indexer/16?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vegarsti and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: client/duneapi/client.go [hunk 1]
```diff
@@ -155,3 +163,123 @@ func (c *client) idempotencyKey(rpcBlock models.RPCBlock) string {
...
+			return err
+		}
+		err = fmt.Errorf("got non-OK response, status code: %s body: %s", responseStatus, responseBody)
+		return err
+	}
+
+	return nil
+}
+
+func (c *client) GetProgressReport(ctx context.Context) (*models.Progress, error) {
```
Comment (vegarsti) [inline client/duneapi/client.go:225]: Feels like there's too many `if err != nil` blocks here, what can I improve?

File: models/progress.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package models
+
+type Progress struct {
```
Comment (msf) [inline models/progress.go:3]: `blockchainIndexProgress` or something?

Identify more things? what's the chain name?

**Comment by vegarsti** [comment]
The failing test is a bit flaky, there's a race condition. The next PR is more stable.

File: client/duneapi/models.go [hunk 3]
```diff
@@ -46,3 +46,16 @@ type BlockchainIngestRequest struct {
 	IdempotencyKey  string
 	Payload         []byte
 }
+
+type BlockchainProgress struct {
+	LastIngestedBlockNumber int64 `json:"last_ingested_block_number"`
+	LatestBlockNumber       int64 `json:"latest_block_number"`
+}
```
Comment (helanto) [inline client/duneapi/models.go:53]: If I the server responds with more fields (i.e. updated_at), this struct will be able to handle it, right ? I mean deser doesn't fail on unknown fields ?

Comment (vegarsti) [reply client/duneapi/models.go:53]: Correct!

**Pull request merged**